### DISCUSSION TITLE
Script Editor shortcuts proposals and analysis.

### DISCUSSION THREAD

**Discussion by setanarut** [godotengine/godot-proposals#4003] [Editor]
Some important settings concerning the `Script Editor` are very difficult to reach. Also it is not possible to assign any keyboard shortcuts. 

1. A new `View` menu should be added to the `Script Editor`.
2. It should be possible to assign a keyboard shortcuts to some settings such as `Word Wrap`, `Show Minimap` etc... ( _Related proposal_ -> https://github.com/godotengine/godot-proposals/issues/1548 ) 

![menu2](https://user-images.githubusercontent.com/36481442/154762877-07f6de34-7306-4a08-a138-fa0d9a182d5e.png)

`Script Editor` shortcuts and `Text Editor` settings are badly named and hierarchically incorrect.

<img width="600" alt="badnaming" src="https://user-images.githubusercontent.com/36481442/154762939-c77c47de-0cc3-42bc-b2d6-ef633949e8ed.png">

Here is a diagram showing the painful steps to follow to enable and disable the `Minimap`. Pay attention to the hierarchy and the steps to be followed according to the above headings.

<img width="352" alt="diagram" src="https://user-images.githubusercontent.com/36481442/154763276-0159834e-9074-464b-aff6-8b075a3892ca.png">

Another wrong ownership example. `External Editor` setting is a setting that belongs to a `superscope Godot Editor` and not to the `Script Editor`(Text Editor?).

<img width="195" alt="Ekran Resmi 2022-02-18 23 38 07" src="https://user-images.githubusercontent.com/36481442/154765263-9b3d8aa5-345c-438b-bca4-cf5754f50948.png">

Thanks.

**Comment by miv391** [comment]
As the Godot 4 now has a command palette (Ctrl+Shift+P in Windows), I would suggest that settings like the mini map should be togglable using the command palette just like in Visual Studio Code.

![image](https://user-images.githubusercontent.com/49998025/154806689-183a3194-03ce-4f3e-9d1a-670893ad1dc5.png)

**Reply by setanarut** [reply]
yes this is more important than the menu and keymap in my opinion.

**Comment by Mickeon** [comment]
I saw this old discussion, and I would like to chime in on this to bring more attention to the issue.

I would avoid adding another option to the top bar, but it seems to be the only way. Perhaps it shouldn't be called View, perhaps it could be a simple icon on the top-right. It should not be just about "things you can view", rather more about "preferences".

One unfortunate problem, even if this were to exist, is that Editor Settings don't always refresh right away. Some may even cause the Editor to stutter. So that needs to be taken into account.
